types/webgl2/index.d.ts

```diff
@@ -387,17 +392,30 @@ interface WebGL2RenderingContext extends WebGLRenderingContext {
     compressedTexImage2D(target: number, level: number, internalformat: number, width: number,
         height: number, border: number, imageSize: number, offset: number): void;
     compressedTexImage2D(target: number, level: number, internalformat: number, width: number,
-        height: number, border: number, srcData: ArrayBufferView | null,
+        height: number, border: number, srcData: Int8Array | Int16Array | Int32Array | Uint8Array | Uint16Array | Uint32Array |
+        Uint8ClampedArray | Float32Array | Float64Array | DataView | null, srcOffset?: number, srcLengthOverride?: number): void;
+    // For compatibility with WebGL 1 context in older Typescript versions.
+    compressedTexImage2D(target: number, level: number, internalformat: number, width: number,
+        height: number, border: number, srcData: ArrayBufferView,
         srcOffset?: number, srcLengthOverride?: number): void;
 
     compressedTexImage3D(target: number, level: number, internalformat: number, width: number,
         height: number, depth: number, border: number, imageSize: number, offset: number): void;
+    compressedTexImage3D(target: number, level: number, internalformat: number, width: number,
```

Comment:
That change shouldn't be required, I think. `compressedTexImage3D` was introduced in WebGL2, so it shouldn't clash with anything in lib.dom.d.ts right now.

If you want to keep the verbose version, please remove `null`. The [specification](https://www.khronos.org/registry/webgl/specs/latest/2.0/) states that `srcData` has to be an `ArrayBufferView` (which excludes `null`).